File: lab2/ex2.py
# Jakub Musiał 268442

import matplotlib.pyplot as plt
import numpy as np
import argparse


def parse_args():
    parser = argparse.ArgumentParser()

    parser.add_argument("-d", "--display", action="store_true")
    parser.add_argument("-s", "--save", type=str, default=None)

    opt = parser.parse_args()
    return vars(opt)


def f(x: np.float64) -> np.float64:
    return np.exp(x) * np.log(np.float64(1) + np.exp(-x))


def main(display: bool, save: str = None):
    print("f(x) = e^x * ln(1 + e^(-x)) plot")

    X = np.linspace(0, 50, num=1000, dtype=np.float64)
    Y = f(X)

    plt.figure(figsize=(7, 5))
    plt.plot(X, Y, )
    plt.xlabel("$x$")
    plt.ylabel("$f(x) = e^x * \ln(1 + e^{-x})$")

    if save:
        plt.savefig(save)

    if display:
        plt.show();


if __name__ == "__main__":
    main(**parse_args())

File: lab2/ex2.jl
# Jakub Musiał 268442

module FuncPlot

using LinearAlgebra
using Plots


function f(x::Float64)::Float64
    return exp(x) * log(Float64(1) + exp(-x))
end


function main(save::Union{String, Nothing})
    println("f(x) = e^x * ln(1 + e^(-x)) plot\n")

    X::Vector{Float64} = LinRange(0.0, 50.0, 1000)
    Y = map(f, X)

    function_plot = plot(
        X, Y,
        label="f(x)", xlabel="x", ylabel="f(x) = e^x * ln(1 + e^(-x))"
    )

    if save === nothing
        display(function_plot)
    else
        Plots.png(function_plot, save)
    end
end

end # FuncPlot


if abspath(PROGRAM_FILE) == @__FILE__
    save = nothing
    if length(ARGS) == 1
        save = ARGS[1]
    end

    FuncPlot.main(save)
end
